src/Elements/tank.cpp

```diff
@@ -107,6 +108,7 @@ void Tank::initialize(Network* nw)
     outflow = 0.0;
     pastOutflow = 0.0;
     quality = initQual;
+    if (qualSource) qualSource->quality = quality;
```

Comment:
initialize the source quality

Reply:
Again, there is no need to initialize `qualSource->quality `here.